test/test-udp-reuseport.c

```diff
@@ -0,0 +1,258 @@
+static unsigned int thread_loop2_recv;
+static unsigned int sent;
+
+static uv_loop_t* main_loop;
+static uv_loop_t* thread_loop1;
+static uv_loop_t* thread_loop2;
+static uv_udp_t thread_handle1;
+static uv_udp_t thread_handle2;
+static uv_timer_t thread_timer_handle1;
+static uv_timer_t thread_timer_handle2;
+
+static void alloc_cb(uv_handle_t* handle,
+                     size_t suggested_size,
+                     uv_buf_t* buf) {
+  buf->base = malloc(suggested_size);
+  buf->len = (int) suggested_size;
+}
+
+static void ticktack(uv_timer_t* timer) {
+  int done = 0;
+
+  ASSERT(timer == &thread_timer_handle1 || timer == &thread_timer_handle2);
+
+  uv_mutex_lock(&mutex);
+  if (received == MAX_UDP_DATAGRAMS) {
+    done = 1;
+  }
+  uv_mutex_unlock(&mutex);
+
+  if (done) {
+    uv_close((uv_handle_t*) timer, NULL);
+    if (timer->loop == thread_loop1)
+      uv_close((uv_handle_t*) &thread_handle1, NULL);
+    if (timer->loop == thread_loop2)
+      uv_close((uv_handle_t*) &thread_handle2, NULL);
+  }
+}
+
+static void on_recv(uv_udp_t* handle,
+                    ssize_t nr,
+                    const uv_buf_t* buf,
+                    const struct sockaddr* addr,
+                    unsigned flags) {
+  ASSERT_OK(flags);
+  ASSERT(handle == &thread_handle1 || handle == &thread_handle2);
+
+  ASSERT_GE(nr, 0);
+
+  if (nr == 0) {
+    ASSERT_NULL(addr);
+    free(buf->base);
+    return;
+  }
+
+  ASSERT_NOT_NULL(addr);
+  ASSERT_EQ(5, nr);
+  ASSERT(!memcmp("Hello", buf->base, nr));
+  free(buf->base);
+
+  if (handle->loop == thread_loop1)
+    thread_loop1_recv++;
+
+  if (handle->loop == thread_loop2)
+    thread_loop2_recv++;
+
+  uv_mutex_lock(&mutex);
+  received++;
+  uv_mutex_unlock(&mutex);
+}
+
+static void on_send(uv_udp_send_t* req, int status) {
+  ASSERT_OK(status);
+  ASSERT_PTR_EQ(req->handle->loop, main_loop);
+
+  if (++sent == MAX_UDP_DATAGRAMS)
+    uv_close((uv_handle_t*) req->handle, NULL);
+}
+
+static void run_event_loop(void* arg) {
+  int r;
+  uv_loop_t* loop = (uv_loop_t*) arg;
+  ASSERT(loop == thread_loop1 || loop == thread_loop2);
+
+  /* Notify the main thread to start sending data. */
+  uv_sem_post(&semaphore);
+  r = uv_run(loop, UV_RUN_DEFAULT);
+  ASSERT_OK(r);
+}
+
+static void bind_socket_and_prepare_recv(uv_loop_t* loop, uv_udp_t* handle) {
+  struct sockaddr_in addr;
+  int r;
+
+  ASSERT_OK(uv_ip4_addr("127.0.0.1", TEST_PORT, &addr));
+
+  r = uv_udp_init(loop, handle);
+  ASSERT_OK(r);
+
+  r = uv_udp_bind(handle, (const struct sockaddr*) &addr, UV_UDP_REUSEPORT);
+  ASSERT_OK(r);
+
+  r = uv_udp_recv_start(handle, alloc_cb, on_recv);
+  ASSERT_OK(r);
+}
+
+TEST_IMPL(udp_reuseport) {
+  struct sockaddr_in addr;
+  uv_buf_t buf;
+  int r;
+  int i;
+
+  r = uv_mutex_init(&mutex);
+
+  r = uv_sem_init(&semaphore, 0);
+  ASSERT_OK(r);
+
+  main_loop = uv_default_loop();
+  ASSERT_NOT_NULL(main_loop);
+
+  /* Create receiving socket per event loop. */
+  thread_loop1 = uv_loop_new();
```

Comment:
Can we avoid using `uv_loop_new()` and `uv_loop_delete()` as they're deprecated

Reply:
Done! Also updated the `test-tcp-reuseport.c`